lib/timers.js

```diff
@@ -554,67 +557,73 @@ function tryOnImmediate(immediate, queue) {
   }
 }
 
+function runCallback(timer) {
```

Comment:
I use this construct in other projects, and found it hard to make generic;
the call semantics don't cover all uses cases well.  I have 3 variants,
this version I used here is a 4th.

That being said, I agree, an optimized call invoker is a nice utility to have.